apps/condo/domains/billing/utils/testSchema/testUtils.js

```diff
@@ -0,0 +1,126 @@
+const { faker  } = require('@faker-js/faker/locale/ru')
+
+const { makeLoggedInAdminClient, makeClient } = require('@open-condo/keystone/test.utils')
+
+
+const { AcquiringTestMixin } = require('@condo/domains/billing/utils/testSchema/mixins/acquiring')
+const { BillingTestMixin } = require('@condo/domains/billing/utils/testSchema/mixins/billing')
+const { ContactTestMixin } = require('@condo/domains/billing/utils/testSchema/mixins/contact')
+const { MeterTestMixin } = require('@condo/domains/billing/utils/testSchema/mixins/meter')
+const { OrganizationTestMixin } = require('@condo/domains/billing/utils/testSchema/mixins/organization')
+const { PropertyTestMixin } = require('@condo/domains/billing/utils/testSchema/mixins/property')
+const { ResidentTestMixin } = require('@condo/domains/billing/utils/testSchema/mixins/resident')
+const {
+    makeClientWithServiceUser,
+    makeClientWithSupportUser,
+    makeLoggedInClient,
+} = require('@condo/domains/user/utils/testSchema')
+
+/**
+ * @mixes BillingTestMixinType
+ * @mixes AcquiringTestMixin
+ * @mixes MeterTestMixin
+ * @mixes ResidentTestMixin
+ * @mixes PropertyTestMixin
+ * @mixes ContactTestMixin
+ */
+class TestUtils {
+    /**
+     * @param {Object[]} [mixins=[]] - An array of mixins to apply
+     */
+    constructor (mixins = []) {
+        this.clients = {}
+        const resolvedMixins = this.resolveMixins(mixins)
+        if (resolvedMixins.length) {
+            Object.assign(this, ...resolvedMixins)
+        }
+        this.mixins = resolvedMixins
+    }
+
+    resolveMixins (mixins) {
+        const resolvedMixins = new Map()
+        const dependencyGraph = new Map()
+        const addMixinToGraph = (mixin) => {
+            if (!resolvedMixins.has(mixin)) {
+                resolvedMixins.set(mixin, mixin)
+                dependencyGraph.set(mixin, mixin.dependsOn || [])
+                if (mixin.dependsOn) {
+                    mixin.dependsOn.forEach(addMixinToGraph)
+                }
+            }
+        }
+        mixins.forEach(addMixinToGraph)
+        const sortedMixins = this.topologicalSort(dependencyGraph)
+        return sortedMixins.map(mixin => resolvedMixins.get(mixin))
+    }
+
+    topologicalSort (dependencyGraph) {
```

Comment:
It'll be better to mark this one as "private method"

Reply:
Done